### PR TITLE
Implement Fake Data Layer

### PR DESCRIPTION
**Pull request by shokeenashish4**
This PR implements a fake data layer that allows a user to verify the App's behavior without having to utilize real data sources.https://github.com/shokeenashish4/figma_shopping_app/assets/150513800/58b10725-8b85-40f7-b3aa-654a22f7bacd

### CODE REVIEW

File: lib/data/products_repository.dart
Comment (shubhamsinghshubham777) [inline lib/data/products_repository.dart:1]: There's no code in this file yet. We can remove it.

File: lib/ui/screens/dashboard_screen.dart [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+  @override
+  State<DashboardScreen> createState() => _DashboardScreenState();
+}
+
+class _DashboardScreenState extends State<DashboardScreen> {
+  int selectedIndex = 0;
+  final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey();
+
+  static const List<Widget> _widgetOptions = <Widget>[
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/dashboard_screen.dart]: Do we really need to make this variable `static`?

File: lib/ui/screens/onboarding/signup_onboarding_screen.dart [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                Future.delayed(
+                  const Duration(seconds: 3),
+                  () {
+                    Navigator.of(context).push(
+                      MaterialPageRoute(
+                        builder: (context) => const GenderScreen(),
+                      ),
+                    );
+                  },
+                );
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/signup_onboarding_screen.dart]: No need to use `Future.delayed` here. Use Future.delayed in `authRepo.signUp`'s implementation 👍🏻

File: lib/ui/screens/onboarding/social_onboarding_screen.dart [hunk 3]
```diff
@@ -42,23 +43,59 @@ class SocialButtonsSection extends StatelessWidget {
...
+              const Duration(seconds: 3),
+              () {
+                Navigator.of(context).push(
+                  MaterialPageRoute(
+                    builder: (context) => const GenderScreen(),
+                  ),
+                );
+              },
+            );
+          },
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/social_onboarding_screen.dart]: While waiting for the response, we can show some kind of indication on the screen [maybe a circular progress indicator or `Loading...` text] (and this applies to all screens throughout the app) such that the user is never left hanging and he always gets some kind of response from the app.

File: lib/ui/screens/onboarding/social_onboarding_screen.dart [hunk 4]
```diff
@@ -42,23 +43,59 @@ class SocialButtonsSection extends StatelessWidget {
...
+            Future.delayed(
+              const Duration(seconds: 3),
+              () {
+                Navigator.of(context).push(
+                  MaterialPageRoute(
+                    builder: (context) => const GenderScreen(),
+                  ),
+                );
+              },
+            );
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/social_onboarding_screen.dart]: We're using this exact same code snippet in both buttons. Better to extract it out as a function instead 👍🏻

File: lib/ui/screens/onboarding/social_onboarding_screen.dart [hunk 5]
```diff
@@ -68,20 +105,24 @@ class SocialButtonsSection extends StatelessWidget {
...
 
   @override
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 5),
       child: ElevatedButton(
-        onPressed: () {},
+        onPressed: () {
+          onTap();
+        },
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/social_onboarding_screen.dart]: We can rewrite this to `onPressed: onTap`

File: lib/ui/screens/onboarding/verification_code_screen.dart [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+          decoration: const InputDecoration(
+            enabledBorder: _inputBorder,
+            focusedBorder: _inputBorder,
+            counterText: "",
+          ),
+          textInputAction: inputAction,
+          keyboardType: TextInputType.number,
+          onChanged: (_) {
+            onSubmitted();
+          },
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/verification_code_screen.dart]: Optional: We can shorten this down to `onChanged: (_) => onSubmitted()`

File: pubspec.yaml [hunk 7]
```diff
@@ -11,6 +11,8 @@ dependencies:
     sdk: flutter
   flutter_svg: ^2.0.9
   awesome_extensions: ^2.0.10
+  get_it: ^7.6.4
+  easy_sidemenu: ^0.5.0
```
Comment (shubhamsinghshubham777) [inline pubspec.yaml]: Are we using any of these dependencies in the code till now? If not, we can remove it 👍🏻

**Review by shubhamsinghshubham777 — CHANGES_REQUESTED** [review]
Please check out the comments @shokeenashish4 and please also add demo videos/screenshots in this PR's description ✅

File: lib/data/products_repository.dart
Comment (shokeenashish4) [reply lib/data/products_repository.dart:1]: done

File: lib/ui/screens/onboarding/social_onboarding_screen.dart (hunk 4 above)
Comment (shokeenashish4) [reply lib/ui/screens/onboarding/social_onboarding_screen.dart]: done.

File: lib/ui/screens/onboarding/social_onboarding_screen.dart (hunk 5 above)
Comment (shokeenashish4) [reply lib/ui/screens/onboarding/social_onboarding_screen.dart]: done.

File: lib/ui/screens/onboarding/verification_code_screen.dart (hunk 6 above)
Comment (shokeenashish4) [reply lib/ui/screens/onboarding/verification_code_screen.dart]: done

File: pubspec.yaml (hunk 7 above)
Comment (shokeenashish4) [reply pubspec.yaml]: removed

File: lib/ui/screens/onboarding/social_onboarding_screen.dart (hunk 3 above)
Comment (shokeenashish4) [reply lib/ui/screens/onboarding/social_onboarding_screen.dart]: done

File: lib/ui/screens/onboarding/signup_onboarding_screen.dart (hunk 2 above)
Comment (shokeenashish4) [reply lib/ui/screens/onboarding/signup_onboarding_screen.dart]: done

File: lib/ui/screens/dashboard_screen.dart (hunk 1 above)
Comment (shokeenashish4) [reply lib/ui/screens/dashboard_screen.dart]: Removed the static and made it a 'final' list instead.

File: lib/ui/screens/onboarding/login_screen.dart [hunk 8]
```diff
@@ -56,7 +56,12 @@ class _LoginScreenState extends State<LoginScreen> {
             () {
               Navigator.of(context).push(
                 MaterialPageRoute(
-                  builder: (context) => const GenderScreen(),
+                  builder: (context) {
+                    const Center(
+                      child: CircularProgressIndicator(),
+                    );
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/login_screen.dart]: Let's remove this as this LOC (Line of code) has no effect.

File: lib/ui/screens/onboarding/new_pwd_screen.dart [hunk 9]
```diff
@@ -49,6 +49,7 @@ class _NewPwdScreenState extends State<NewPwdScreen> {
               const Duration(seconds: 2),
               () {
                 if (wasLoggedIn) {
+                  const Center(child: CircularProgressIndicator());
```
Comment (shubhamsinghshubham777) [inline lib/ui/screens/onboarding/new_pwd_screen.dart]: Same. Let's remove this as well.

**Review by shubhamsinghshubham777 — CHANGES_REQUESTED** [review]

**Comment by shokeenashish4** [comment]
Fixed the code with the changes you recommended

**Review by shubhamsinghshubham777 — APPROVED** [review]

**Pull request merged**